src/main/resources/i18n/Scalars_de.properties

```diff
@@ -0,0 +1,32 @@
+#
+# This resource bundle is used for the scalar code to produce i18n messages
+#
+# The keys have the format of rule class name and then message type within that.  Most rules
+# will only have 1 or 2 message keys
+#
+# Please try and keep this sorted within rule class and use # between sections so the IDEA Ctrl-Alt-L reformat does not bunch
+# them too tightly.
+#
+# REMEMBER - a single quote ' in MessageFormat means things that are never replaced within them
+# so use 2 '' characters to make it one ' on output.  This will take for the form ''{0}''
+#
+# Prior to Java 9, properties files are encoded in ISO-8859-1.
+# We have to use \u00fc instead of the German ue character, \u00e4 for ae, \u00f6 for oe, \u00df for ss
+#
+Scalar.unexpectedAstType=Erwartet wurde ein AST-Typ von ''{0}'', aber es war ein ''{1}''
```

Comment:
AST Type should not be translated imho

Reply:
Sofern man den AST-Type doch übersetzen wollen würde, wären folgende Optionen denkbar.

- Erwartet wurde ein AST-Typ vom Typ X, jedoch war dieser Typ ein Y
- Erwartet wurde DER AST-Typ X, es kam aber ein AST-Typ Y.

Sofern man den - wie von @andimarek vorgeschlagen - AST-Typ weglassen möchte, könnte man das ganze so schreiben:

- Erwartet wurde X, gekommen ist aber Y 

Was will ich damit sagen? Dass das "es war ein....." nicht passt, da das Wort "erwarten" in diesem Zusammenhang ein "kommen" fordert.

Ich erwartete @dondonz , es kam aber @andimarek 
Ich erwartete @dondonz , es war aber @andimarek (, der gekommen ist) 
Ohne das, was in Klammern steht, würde der Satz so nicht funktionieren.

Etwas unkritischer ist folgendes zu sehen, wenn ich vorher noch einen abstrakten Klassifizierer einbaue.

Ich erwartete die Person @dondonz , aber die Person war @andimarek (,die dafür sorgte, dass meine Erwartungshaltung nicht eingehalten worden ist^^)
In dem Fall bezieht sich das "aber (die Person)" nur auf das Unterobjekt von Person aus dem vorhergehenden Hauptsatz.

Soviel zu dem Versuch, mein Sprachgefühl etwas zu Papier zu bekommen.

Reply:
Hi @knotenpunkt, thanks for your comments on this PR, I really do value having experts review this PR.

I thought German grammar was hard already but Denglisch grammar is beyond me! All these English words really mess with the grammar 😅. While I will be merging this PR as is soon, wanted to say I'm going to ask for more reviews of these strings and I will take your comments into account. I didn't want to block the PR while I ask for more reviews.

Forgive me for replying in English!